example-async.js

```diff
@@ -0,0 +1,158 @@
+'use strict'
+
+const util = require('util')
+const Fastify = require('fastify')
+
+function build (opts) {
+  const fastify = Fastify(opts)
+
+  fastify
+    .register(require('fastify-jwt'), { secret: 'supersecret' })
+    .register(require('fastify-leveldb'), { name: 'authdb-async' })
+    .register(require('./fastify-auth'))
+    .after(routes)
+
+  fastify.decorate('verifyJWTandLevel', verifyJWTandLevel)
+  fastify.decorate('verifyUserAndPassword', verifyUserAndPassword)
+
+  function verifyJWTandLevel (request, reply, done) {
+    const jwt = this.jwt
+    const level = this.level
+
+    if (request.body && request.body.failureWithReply) {
+      reply.code(401).send({ error: 'Unauthorized' })
+      return done(new Error())
```

Comment:
I think it’s better if you do a return Promise.reject(err). If you use the Unhauthorized error class from http-errors, there is no need to call reply.code(401).

Reply:
regarding the `http-errors` - do you mean I should add https://www.npmjs.com/package/http-errors as dependency?

Reply:
given that it seems not to be a requirement and such small use case it not yet enough for me to add another dev dependency - I'll just resolve this discussion